lib/Site.js

```diff
@@ -185,6 +188,39 @@ Site.prototype.collectBaseUrl = function () {
   return Promise.resolve();
 };
 
+/**
+ * Collects the user defined variables map in the site/subsites
+ */
+Site.prototype.collectUserDefinedVariablesMap = function () {
+  // The key is the base directory of the site/subsites,
+  // while the value is user defined variable map for that particular subsite
+  this.userDefinedVariablesMap = {};
+
+  Object.keys(this.baseUrlMap).forEach((base) => {
+    const userDefinedVariables = {};
+    const definedVariablePath = path.resolve(base, USER_VARIABLES_PATH);
+    let content;
+    try {
+      content = fs.readFileSync(definedVariablePath, 'utf8');
+    } catch (e) {
+      content = '';
+      logger.info(e.message);
```

Comment:
Let's elevate this to `logger.warn` and create the file by default in `Site.initSite`.